adafruit_gps.py

```diff
@@ -200,6 +201,7 @@ def _parse_sentence(self):
                 actual ^= ord(sentence[i])
             if actual != expected:
                 return None  # Failed to validate checksum.
+            self.sentence = sentence
```

Comment:
great idea! we'd want a new private variable that gets set, and isn't changed (below, the checksum is removed) and then a property getter

Reply:
@ladyada thanks for the comment. I split out `_parse_sentence` into `_read_sentence` which will set the property `raw_sentence`. I've also added a new property getter that returns the `raw_sentence` string.

Reply:
Looks like you made the requested changes. I'll be testing the examples, and will merge the PR assuming they pass

Reply:
Ok, so for some reason, when self.raw_sentence is defined as None in __init__, it says:
`AttributeError: 'GPS' object has no attribute 'raw_sentence'`
The same thing happens with all example files except the datalogging files, which don't use the gps module.

When I comment out `self.raw_sentence = None`, that fixes it.

Reply:
Ok, so I think that you can't have both a variable defined in \_\_init\_\_ as None and a property getter. @ladyada Would it be ok if the variable got removed from \_\_init\_\_? I've tested with the example files, and that seems to be the only way to get it to run.